spec/lib/rex/proto/kerberos/model/ticket_enc_part_spec.rb

```diff
@@ -0,0 +1,347 @@
+      encryption_key.type = 23
+      encryption_key.value = "KUJQCsERWUdIKHkx"
+      encryption_key
+    end
+
+    let(:cname) do
+      principal_name = Rex::Proto::Kerberos::Model::PrincipalName.new
+      principal_name.name_type = 1
+      principal_name.name_string = ["fake_mysql"]
+      principal_name
+    end
+
+    let(:transited) do
+      transited_encoding = Rex::Proto::Kerberos::Model::TransitedEncoding.new
+      transited_encoding.tr_type = 0
+      transited_encoding.contents = ""
+      transited_encoding
+    end
+
+    let(:sample_auth_data) do
+      "0\x82\x02\x7F0\x82\x02{\xA0\x03\x02\x01\x01\xA1\x82\x02r\x04\x82\x02n0\x82\x02j0\x82\x02f\xA0\x04\x02\x02\x00\x80\xA1\x82\x02\\\x04\x82\x02X\x04\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\xBC\x01\x00\x00H\x00\x00\x00\x00\x00\x00\x00\n\x00\x00\x00\x1E\x00\x00\x00\b\x02\x00\x00\x00\x00\x00\x00\x06\x00\x00\x00\x14\x00\x00\x00(\x02\x00\x00\x00\x00\x00\x00\a\x00\x00\x00\x14\x00\x00\x00@\x02\x00\x00\x00\x00\x00\x00\x01\x10\b\x00\xCC\xCC\xCC\xCC\xAC\x01\x00\x00\xCC\xCC\xCC\xCCl\\\x00\x00\x80\x94\x11\f\x92\x84\xD8\x01\xFF\xFF\xFF\xFF\xFF\xFF\xFF\x7F\xFF\xFF\xFF\xFF\xFF\xFF\xFF\x7F\x80\x94\x11\f\x92\x84\xD8\x01\x00\x00\x00\x00\x00\x00\x00\x00\xFF\xFF\xFF\xFF\xFF\xFF\xFF\x7F\x14\x00\x14\x00\x97\xDF\x00\x00\x00\x00\x00\x00`\xE2\x00\x00\x00\x00\x00\x00\x84\x9D\x00\x00\x00\x00\x00\x00\x8C;\x00\x00\x00\x00\x00\x00\x94\xC2\x00\x00\x00\x00\x00\x00~\xC9\x00\x00\xF4\x01\x00\x00\xF4\x01\x00\x00\x01\x02\x00\x00\x05\x00\x00\x00r4\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x83\x88\x00\x00\x10\x00\x10\x00\x9C\x94\x00\x00\xA2\xD7\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x10\x02\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\n\x00\x00\x00\x00\x00\x00\x00\n\x00\x00\x00f\x00a\x00k\x00e\x00_\x00m\x00y\x00s\x00q\x00l\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\x00\x00\x01\x02\x00\x00\a\x00\x00\x00\x00\x02\x00\x00\a\x00\x00\x00\b\x02\x00\x00\a\x00\x00\x00\x06\x02\x00\x00\a\x00\x00\x00\a\x02\x00\x00\a\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\b\x00\x00\x00\x00\x00\x00\x00\b\x00\x00\x00D\x00W\x00.\x00L\x00O\x00C\x00A\x00L\x00\x05\x00\x00\x00\x01\x05\x00\x00\x00\x00\x00\x05\x15\x00\x00\x00\x02\x87ZU\xFDf{\xCC\xF7\xBC\xAF\x16\xF4\x01\x00\x00\x00\x00\x00\x00\x80\x94\x11\f\x92\x84\xD8\x01\x14\x00f\x00a\x00k\x00e\x00_\x00m\x00y\x00s\x00q\x00l\x00\x00\x00v\xFF\xFF\xFF\xE2X\xD7\xF5\x06\x17d\x9Cw\xC6\xFF\x16\x13\x16]\xE3\x00\x00\x00\x00v\xFF\xFF\xFF\xAFd\xEA\x1D\x9F\xA0c\xE9\xB1\x04\xD2S\xC4\x82\xCEa\x00\x00\x00\x00"
```

Comment:
I don't know what other folk have been doing, but I've been using this quick hack to tidy up the blobs:

```ruby
require 'active_support/core_ext/array'; puts data.bytes.map { |x| "\\x#{x.to_s(16).rjust(2, '0')}" }.in_groups_of(16).map { |row| "\"#{row.join("")}\"" }.join(" \\ \n")
```

```suggestion
      "\x30\x82\x02\x7f\x30\x82\x02\x7b\xa0\x03\x02\x01\x01\xa1\x82\x02" \ 
      "\x72\x04\x82\x02\x6e\x30\x82\x02\x6a\x30\x82\x02\x66\xa0\x04\x02" \ 
      "\x02\x00\x80\xa1\x82\x02\x5c\x04\x82\x02\x58\x04\x00\x00\x00\x00" \ 
      "\x00\x00\x00\x01\x00\x00\x00\xbc\x01\x00\x00\x48\x00\x00\x00\x00" \ 
      "\x00\x00\x00\x0a\x00\x00\x00\x1e\x00\x00\x00\x08\x02\x00\x00\x00" \ 
      "\x00\x00\x00\x06\x00\x00\x00\x14\x00\x00\x00\x28\x02\x00\x00\x00" \ 
      "\x00\x00\x00\x07\x00\x00\x00\x14\x00\x00\x00\x40\x02\x00\x00\x00" \ 
      "\x00\x00\x00\x01\x10\x08\x00\xcc\xcc\xcc\xcc\xac\x01\x00\x00\xcc" \ 
      "\xcc\xcc\xcc\x6c\x5c\x00\x00\x80\x94\x11\x0c\x92\x84\xd8\x01\xff" \ 
      "\xff\xff\xff\xff\xff\xff\x7f\xff\xff\xff\xff\xff\xff\xff\x7f\x80" \ 
      "\x94\x11\x0c\x92\x84\xd8\x01\x00\x00\x00\x00\x00\x00\x00\x00\xff" \ 
      "\xff\xff\xff\xff\xff\xff\x7f\x14\x00\x14\x00\x97\xdf\x00\x00\x00" \ 
      "\x00\x00\x00\x60\xe2\x00\x00\x00\x00\x00\x00\x84\x9d\x00\x00\x00" \ 
      "\x00\x00\x00\x8c\x3b\x00\x00\x00\x00\x00\x00\x94\xc2\x00\x00\x00" \ 
      "\x00\x00\x00\x7e\xc9\x00\x00\xf4\x01\x00\x00\xf4\x01\x00\x00\x01" \ 
      "\x02\x00\x00\x05\x00\x00\x00\x72\x34\x00\x00\x00\x00\x00\x00\x00" \ 
      "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" \ 
      "\x00\x00\x00\x83\x88\x00\x00\x10\x00\x10\x00\x9c\x94\x00\x00\xa2" \ 
      "\xd7\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x10\x02\x00\x00\x00" \ 
      "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" \ 
      "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" \ 
      "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x0a" \ 
      "\x00\x00\x00\x00\x00\x00\x00\x0a\x00\x00\x00\x66\x00\x61\x00\x6b" \ 
      "\x00\x65\x00\x5f\x00\x6d\x00\x79\x00\x73\x00\x71\x00\x6c\x00\x00" \ 
      "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" \ 
      "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" \ 
      "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" \ 
      "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\x00\x00\x01" \ 
      "\x02\x00\x00\x07\x00\x00\x00\x00\x02\x00\x00\x07\x00\x00\x00\x08" \ 
      "\x02\x00\x00\x07\x00\x00\x00\x06\x02\x00\x00\x07\x00\x00\x00\x07" \ 
      "\x02\x00\x00\x07\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" \ 
      "\x00\x00\x00\x08\x00\x00\x00\x00\x00\x00\x00\x08\x00\x00\x00\x44" \ 
      "\x00\x57\x00\x2e\x00\x4c\x00\x4f\x00\x43\x00\x41\x00\x4c\x00\x05" \ 
      "\x00\x00\x00\x01\x05\x00\x00\x00\x00\x00\x05\x15\x00\x00\x00\x02" \ 
      "\x87\x5a\x55\xfd\x66\x7b\xcc\xf7\xbc\xaf\x16\xf4\x01\x00\x00\x00" \ 
      "\x00\x00\x00\x80\x94\x11\x0c\x92\x84\xd8\x01\x14\x00\x66\x00\x61" \ 
      "\x00\x6b\x00\x65\x00\x5f\x00\x6d\x00\x79\x00\x73\x00\x71\x00\x6c" \ 
      "\x00\x00\x00\x76\xff\xff\xff\xe2\x58\xd7\xf5\x06\x17\x64\x9c\x77" \ 
      "\xc6\xff\x16\x13\x16\x5d\xe3\x00\x00\x00\x00\x76\xff\xff\xff\xaf" \ 
      "\x64\xea\x1d\x9f\xa0\x63\xe9\xb1\x04\xd2\x53\xc4\x82\xce\x61\x00" \ 
      "\x00\x00\x00"
```